### ISSUE TITLE
Tunnel Config Copy-er

### ISSUE THREAD

**Issue by met4000** [CyclopsMC/IntegratedDynamics#859]
#### Issue type:

- :heavy_plus_sign: Feature request

____

#### Short description:
Feature request to add an item vaguely like the "memory card" in AE2, or the "conduit probe" in EnderIO; interact with one tunnel (shift right click or something) to copy config, and interact with another (right click?) to paste config. Config refers to any settings that you can change by right-clicking to open the UI, and any variable cards (such as predicates) placed in importers/exporters. 

When pasting config that 'includes' variable cards, a blank variable card would be removed from the player's inventory (or fail with a ig-chat warning/explanation about 'not enough blank cards') and a copy of the original card would be placed in the exporter/importer. Any cards currently in the importer/exporter would be ejected onto the ground in item drop form.

>Kyiki Nasoka: [...] to copy general settings like "facing = Up, Priority = 3, tickDelay = 20"
>Kyiki Nasoka: oh like an item exporter with a Predicate card, copy it's settings and it would duplciate that card and then automatically push it inot the next one?
> met4000: (and take an empty variable card from your inventory)

([link to discord conversation](https://discordapp.com/channels/386052815128100865/386054223688630272/709765822180098088))

**Comment by rubensworks** [comment]
Good idea. Not sure if I'll ever get to implementing this, so PRs are welcome!

**Comment by Jack-McKalling** [comment]
This would already be awesome if it would not touch any variable card but only modify the aspect properties that have variable cards in them (in the original tunnel that was copied). Or just all modified aspects (although might be a little redundant)

**Comment by Archengius** [comment]
Possible implementation of the feature: https://github.com/Archengius/IntegratedDynamics/tree/feature-settings-copier

## Summary

### Gameplay

- Added a new tool: Settings Copier
- Crafting recipe: same as wrench, but with redstone replaced with paper (7 crystallized menril chunks and 1 paper)
- Shift + Right Click on a supported block or a part to save its settings to the tool
- Right Click on another block or part to apply the settings stored in the tool to the block/part
- Settings can only be pasted to the same block type/part type they were originally copied from.
- Pasting completely overrides settings of the target (including aspect properties, offsets and part settings for parts)
- Variable Cards are copied when they are pasted.
- Cards stored in Variable Store are copied as one logical unit, e.g. references from the variables stored in a copied Variable Store to the other variables stored in the same Variable Store will be replaced with references to copied variables (see #1495 on why this is a desired behavior; references to external variables are left unchanged)
- Settings Copier can be cleared in the crafting grid from the settings stored in it.

### Implementation

- Settings Copier uses a new block/part capability ISettingsCopyable to copy/paste settings
- Copied block/part settings are represented by objects implementing IBlockSettings interface. Block Settings are stored in a data component on the Settings Copier and hold a complete state of the block at the time the settings were copied (so if the block/part was changed after the settings were copied, the settings stored in the copier remain unchanged)
- Each block settings class must have an unique type registered in the block settings type registry. Block Settings Type describes how block settings objects are serialized/deserialized to/from NBT.
- Capability is implemented individually for Variable Store, Delay, Materializer and Proxy (with some shared logic in active variable base block entity)
- Base part state implements capability if either offsets or part settings can be changed, and by default part settings copied/pasted include target side, channel, priority, update interval, offsets, offset variables, amount of installed offset upgrades, and modified aspect properties.
- Active variable base-derived parts also implement the capability to allow copying and pasting active variable card and active aspect on top of the settings supported by the base part state.
- Feature has its own gameplay test to test the implementation behavior
